File: interval.NET/IntervalDouble.cs
using IntervalDotNET;

namespace interval.NET
{
    public class IntervalDouble
    {
        internal readonly IntervalDoubleWrapper _interval;

        public double Upper => _interval.Upper;
        public double Lower => _interval.Lower;

        public IntervalDouble(IntervalDoubleWrapper v)
        {
            _interval = v;
        }

        public IntervalDouble(double v) 
        {
            _interval = new IntervalDoubleWrapper(v);
        }

        public IntervalDouble(double lower, double upper)
        {
            _interval = new IntervalDoubleWrapper(lower, upper);
        }

        public override string ToString()
        {
            string? format = "N15";
            format = null;
            return $"Median: {((_interval.Lower + _interval.Upper) / 2).ToString(format)} Error: {((_interval.Upper - _interval.Lower) / 2).ToString(format)} Lower: {_interval.Lower.ToString(format)} Upper: {_interval.Upper.ToString(format)}";
        }        

        public static IntervalDouble operator +(IntervalDouble a, IntervalDouble b)
        {
            return new IntervalDouble(IntervalDoubleWrapper.Add(a._interval, b._interval));
        }

        public static IntervalDouble operator -(IntervalDouble a, IntervalDouble b)
        {
            return new IntervalDouble(IntervalDoubleWrapper.Subtract(a._interval, b._interval));
        }

        public static IntervalDouble operator *(IntervalDouble a, IntervalDouble b)
        {
            return new IntervalDouble(IntervalDoubleWrapper.Multiply(a._interval, b._interval));
        }

        public static IntervalDouble operator /(IntervalDouble a, IntervalDouble b)
        {
            return new IntervalDouble(IntervalDoubleWrapper.Divide(a._interval, b._interval));
        }

        public static IntervalDouble Pow(IntervalDouble x, int n)
        {
            return new IntervalDouble(IntervalDoubleWrapper.Pow(x._interval, n));
        }
    }
}

File: interval.NET.Wrapper/IntervalDouble.cpp
#include "IntervalDouble.hpp"
namespace IntervalDotNET
{
	IntervalDouble::IntervalDouble(double v) : ManagedObject(new Core::IntervalDoubleCore(v)) { }

	IntervalDouble::IntervalDouble(double lower, double upper) : ManagedObject(new Core::IntervalDoubleCore(lower, upper)) { }

	IntervalDouble::IntervalDouble(Core::IntervalDoubleCore* core) : ManagedObject(core) { }

	double IntervalDouble::Median::get()
	{
		return Core::IntervalDoubleCore::Median(m_Instance);
	}

	double IntervalDouble::Error::get()
	{
		return Core::IntervalDoubleCore::Width(m_Instance) / 2;
	}

	String^ IntervalDouble::ToString()
	{
		return String::Format("Median: {0}, Error: {1}", Median, Error);
	}

	IntervalDouble^ IntervalDouble::operator+(IntervalDouble^ x, IntervalDouble^ y)
	{
		return gcnew IntervalDouble(Core::IntervalDoubleCore::Add(x->GetInstance(), y->GetInstance()));
	}

	IntervalDouble^ IntervalDouble::operator-(IntervalDouble^ x, IntervalDouble^ y)
	{
		return gcnew IntervalDouble(Core::IntervalDoubleCore::Subtract(x->GetInstance(), y->GetInstance()));
	}

	IntervalDouble^ IntervalDouble::operator*(IntervalDouble^ x, IntervalDouble^ y)
	{
		return gcnew IntervalDouble(Core::IntervalDoubleCore::Multiply(x->GetInstance(), y->GetInstance()));
	}

	IntervalDouble^ IntervalDouble::operator/(IntervalDouble^ x, IntervalDouble^ y)
	{
		return gcnew IntervalDouble(Core::IntervalDoubleCore::Divide(x->GetInstance(), y->GetInstance()));
	}

	bool IntervalDouble::operator<(IntervalDouble^ x, IntervalDouble^ y)
	{
		return Core::IntervalDoubleCore::LessThan(x->GetInstance(), y->GetInstance());
	}

	bool IntervalDouble::operator<=(IntervalDouble^ x, IntervalDouble^ y)
	{
		return Core::IntervalDoubleCore::LessThanOrEqual(x->GetInstance(), y->GetInstance());
	}

	bool IntervalDouble::operator>(IntervalDouble^ x, IntervalDouble^ y)
	{
		return Core::IntervalDoubleCore::GreaterThan(x->GetInstance(), y->GetInstance());
	}

	bool IntervalDouble::operator>=(IntervalDouble^ x, IntervalDouble^ y)
	{
		return Core::IntervalDoubleCore::GreaterThanOrEqual(x->GetInstance(), y->GetInstance());
	}	

	bool IntervalDouble::Subset(IntervalDouble^ innerInterval, IntervalDouble^ outerInterval)
	{
		return Core::IntervalDoubleCore::Subset(innerInterval->m_Instance, outerInterval->m_Instance);
	}

	IntervalDouble^ IntervalDouble::Pow(IntervalDouble^ x, int n)
	{
		return gcnew IntervalDouble(Core::IntervalDoubleCore::Pow(x->m_Instance, n));
	}

	IntervalDouble^ IntervalDouble::Sqrt(IntervalDouble^ x)
	{
		return gcnew IntervalDouble(Core::IntervalDoubleCore::Sqrt(x->m_Instance));
	}
}